hilti/toolchain/src/compiler/optimizer.cc

```diff
@@ -421,29 +304,51 @@ struct FunctionVisitor : OptimizerVisitor, visitor::PreOrder<bool, FunctionVisit
     }
 
     result_t operator()(const operator_::struct_::MemberCall& x, position_t p) {
-        auto function_id = getID(x, p);
+        if ( ! x.hasOp1() )
+            return false;
+
+        assert(x.hasOp0());
+
+        auto type = x.op0().type();
+        while ( type::isReferenceType(type) )
```

Comment:
I'd be curious to know if we still need this.

Reply:
Looks like this can go now. We still need the `innermostType` helper though.